client/my-sites/plans-v2/products-column/index.tsx

```diff
@@ -56,30 +56,46 @@ const ProductsColumn = ( {
 	// - Owned products from a direct purchase.
 	// - Included as a feature of an owned plan.
 	// - Generic/option product bundles with subtypes.
-	const productObjects: SelectorProduct[] = useMemo(
-		() =>
-			// Convert product slugs to ProductSelector types.
-			[ ...new Set( [ ...ownedProducts, ...includedInPlanProducts, ...SELECTOR_PRODUCTS ] ) ]
-				.map( slugToSelectorProduct )
-				// Remove products that don't fit the filters or have invalid data.
-				.filter(
-					( product: SelectorProduct | null ): product is SelectorProduct =>
-						!! product &&
-						duration === product.term &&
-						PRODUCTS_TYPES[ productType ].includes( product.productSlug ) &&
-						// Don't include a generic/option card if the user already owns a subtype.
-						! ownedProducts.some( ( ownedProduct ) => product.subtypes.includes( ownedProduct ) ) &&
-						// Don't include a generic/option card if the product is included in the owned plan.
-						! includedInPlanProducts.some( ( includedProduct ) =>
-							product.subtypes.includes( includedProduct )
-						)
-				)
-				.map( ( product: SelectorProduct ) => ( {
-					...product,
-					description: getJetpackDescriptionWithOptions( product ),
-				} ) ),
-		[ duration, includedInPlanProducts, ownedProducts, productType ]
-	);
+	const productObjects: SelectorProduct[] = useMemo( () => {
+		// Owned products (plans are filtered out).
+		const ownedProducts = siteProducts
+			.map( ( { productSlug } ) => productSlug )
+			.filter( ( productSlug ) => JETPACK_PRODUCTS_LIST.includes( productSlug ) )
+			.map( slugToSelectorProduct )
+			.filter( ( product ): product is SelectorProduct => product !== null );
```

Comment:
Now owned products are always displayed. They don't go through the filtering process the rest of the cards do.